app/views/upload.scala.html

```diff
@@ -24,6 +25,9 @@ <h2 class="govuk-error-summary__title" id="error-summary-title">
                 </ul>
             </div>
         </div>
+        @form(routes.UploadController.uploadRedirect(consignmentId), Symbol("id") -> "upload-data-form") {
```

Comment:
I would suggest moving this form to after the main upload form, to give some indication that this form is used after the main form. Probably should add a comment about why it is being used as well

Have you tried to call the RecordController.recordsPage(...) directly here, instead of calling the UploadController, which then redirects to the RecordsController.recordsPage(...). So you would have something looking like this:

@form(routes.RecordsController.recordsPage(consignmentId)) {@CRSF.formField}

Reply:
I've given this a go and since it's now a GET rather than a POST we don't need the CSRF token :+1: